src/compile/compile.ts

```diff
@@ -86,7 +87,8 @@ function assembleTopLevelModel(model: Model, topLevelProperties: TopLevelPropert
     ...(style? {style} : {}),
     data: [].concat(
       model.assembleSelectionData([]),
-      model.assembleData()
+      // only assemble data in the root
+      assembleData(model.component.data)
```

Comment:
Yay.  I like that you move it here.  

Maybe rename to `assembleRootData` ? (So we have clear names for both `assembleRootData` and `assembleFacetData`)